nrf_rpc/nrf_rpc_cbor.c

```diff
@@ -13,39 +13,10 @@
 #include <nrf_rpc_cbor.h>
 #include <nrf_rpc_common.h>
 
-struct handler_proxy_ctx {
-	CborValue value;
-	CborParser parser;
-	struct cbor_buf_reader reader;
-	const uint8_t *in_packet;
-};
-
-/* All Context structures are independently defined for API user convenient, but
- * they need the same layout to allow casting between them. Following asserts
- * ensures that.
- */
-NRF_RPC_STATIC_ASSERT(offsetof(struct nrf_rpc_cbor_ctx, encoder) ==
-		      offsetof(struct nrf_rpc_cbor_rsp_ctx, encoder),
-		      "Context structures fields does not match");
-NRF_RPC_STATIC_ASSERT(offsetof(struct nrf_rpc_cbor_ctx, out_packet) ==
-		      offsetof(struct nrf_rpc_cbor_rsp_ctx, out_packet),
-		      "Context structures fields does not match");
-NRF_RPC_STATIC_ASSERT(offsetof(struct nrf_rpc_cbor_ctx, writer) ==
-		      offsetof(struct nrf_rpc_cbor_rsp_ctx, writer),
-		      "Context structures fields does not match");
-
-NRF_RPC_STATIC_ASSERT(offsetof(struct handler_proxy_ctx, value) ==
-		      offsetof(struct nrf_rpc_cbor_rsp_ctx, value),
-		      "Context structures fields does not match");
-NRF_RPC_STATIC_ASSERT(offsetof(struct handler_proxy_ctx, parser) ==
-		      offsetof(struct nrf_rpc_cbor_rsp_ctx, parser),
-		      "Context structures fields does not match");
-NRF_RPC_STATIC_ASSERT(offsetof(struct handler_proxy_ctx, reader) ==
-		      offsetof(struct nrf_rpc_cbor_rsp_ctx, reader),
-		      "Context structures fields does not match");
-NRF_RPC_STATIC_ASSERT(offsetof(struct handler_proxy_ctx, in_packet) ==
-		      offsetof(struct nrf_rpc_cbor_rsp_ctx, in_packet),
-		      "Context structures fields does not match");
+static inline size_t nrf_rpc_cbor_data_len(const struct nrf_rpc_cbor_ctx *ctx)
+{
+	return ctx->zs->payload_mut - ctx->packet;
```

Comment:
What is a difference between `ctx->zs->payload_mut` and `ctx->zs->payload`  ?

Reply:
Union, same thing. One has const as far as I remember.